types/src/system/auction/seigniorage_recipient.rs

```diff
@@ -42,13 +42,18 @@ impl SeigniorageRecipient {
     }
 
     /// Calculates total stake, including delegators' total stake
-    pub fn total_stake(&self) -> U512 {
-        self.stake + self.delegator_total_stake()
+    pub fn total_stake(&self) -> Option<U512> {
+        self.delegator_total_stake()
+            .and_then(|stake| self.stake.checked_add(stake))
```

Comment:
Nit: I see there are more uses of `?` already around here, so maybe it will fit here too? 

```suggestion
        Some(self.delegator_total_stake()?.checked_add(self.stake)?)
```

Reply:
Applied suggestion